lib/utils/analytics/generatePayload.js

```diff
@@ -85,5 +93,11 @@ module.exports = async (serverless) => {
     npmDependencies,
     triggeredDeprecations: Array.from(triggeredDeprecations),
     versions,
+    createdAt: Date.now(),
+    ciName: ci.name,
```

Comment:
I also realized one thing - it's possible to detect that the app is running in CI, but there might be no `ci.name` detected - should we in such case introduce separate properties on e.g. use `ciName: 'unknown'` to signify that it's in CI but we don't know the name? I prefer this approach but I can also see a reason to go with separate propetrties and I'd love to know your opinion @medikoo

Reply:
> I also realized one thing - it's possible to detect that the app is running in CI, but there might be no ci.name detected

Is such case documented by `ci-info` ? I thought that _name_ is unconditionally set

Reply:
I also though it's the case but I've looked at the implementation and `name` is set if it's identified, but it's possible that there's `isCi` set to `true` but `name` couldn't be identified so it's empty.

Reply:
It is also indeed documented: https://github.com/watson/ci-info#ciisci

Reply:
I think setting here _unknown_ is fine. Still it'll be good to also resonate here with eventual handling of _not sure if CI_ cases, when we will propagate data to mixpanel.

So question, is how over there will we communicate record for even which we don't know if was send in CI context or not, and event which we know that was send in CI context, but do not know the name of CI.

How do you think we will handle that?

Reply:
I was thinking like this
- If there's a concrete CI name, then we know which CI engine was used
- If there is `unknown`, then we know that it was executed in CI engine, but it wasn't possible to identify which one
- If there's `untracked` as value, then we don't know if it was executed in CI engine or not, but it's also quite likely that the events from older versions are likely not coming from CI engines (it's disabled in `utils` as of right now)
- If there's `null`, then we assume that it wasn't executed in CI engine

What do you think? Maybe we should introduce a separate property to not pack a lot of different meanings into `ciName`?

Reply:
I've updated the PR with the approach I proposed above, please let me know what do you think here